schema.sql

```diff
@@ -26,6 +26,7 @@ CREATE TABLE IF NOT EXISTS Beekeeper.nodes_log (
 CREATE TABLE IF NOT EXISTS Beekeeper.nodes_history (
     `id`                    VARCHAR(16) NOT NULL,  /*  typically MAC address of the "main" device  */
     `timestamp`             TIMESTAMP(0) NOT NULL,
+    `vsn`                   VARCHAR(64),
```

Comment:
1. Good question. I believe the same way you guys added the `wes_deploy_event` column.
2. Similar to the deploy_wes service. Just fetch it from the node.

Reply:
1. can you investigate if we need versions in our tables and a "upgrade" plan.  i thought i remember (in past projects) there being code that could run when upgrading a table
2. yeah we can just do it in post for every node.

Reply:
1. When this commit was made: https://github.com/waggle-sensor/beekeeper/commit/5497810a6a57fd04f59653012aea99af00138efc#diff-939acd018b931292330e519becd8d003ed0dd20c5e42de62ec3d37553698739a , beekeeper was already deployed, correct?

Reply:
I have no idea.  but we should establish a process for how we will be upgrading our database(s) in production?  to ensure we dont blow away something.

ref: https://dev.mysql.com/doc/refman/5.7/en/mysql-upgrade.html

Reply:
This would fall under "migrations". In this case, we'd need to do an ALTER TABLE add this field.

Unfortunately, this is generally a challenging thing, especially when all the DB abstraction is written by hand...

ORMs do a lot of work to make this possible. For example, in frameworks like Django, as you update your database models, it generates the steps to migrate the tables safely and covers stuff like adding new columns, prompting you to backfill a new required column, preventing bad types, etc.

All that's to say, it's a nontrivial thing we need to think about.

Reply:
This would fall under "migrations". In this case, we'd need to do an ALTER TABLE add this field.

Unfortunately, this is generally a challenging thing, especially when all the DB abstraction is written by hand...

ORMs do a lot of work to make this possible. For example, in frameworks like Django, as you update your database models, it generates the steps to migrate the tables safely and covers stuff like adding new columns, prompting you to backfill a new required column, preventing bad types, etc.

All that's to say, it's a nontrivial thing we need to think about.

Reply:
To be totally honest... part of me wonders if we should consider porting some of this layer to a framework like Django to help abstract details like this automatically.

We can maintain all the APIs as they are now, but it may pay off in the long run.

Reply:
ORM will be ideal for this scenario. I can put together something to just show the data model.

I will prefer to use FastAPI(https://fastapi.tiangolo.com/) instead of django.

Reply:
You mean something like sqlalchemy? (Since I don't think FastAPI does database / model management on its own?)

It's your call. I really love FastAPI... but for a (potentially) very CRUD type app, it seems like Django + Django REST Framework _might_ be a better fit.

Especially with things like Django's builtin admin UI.

Reply:
It uses sqlalchemy in the backend for ORM in fastapi: https://fastapi.tiangolo.com/tutorial/sql-databases/#orms

Reply:
Taken note of this for future work